zerver/tests/fixtures/markdown_test_cases.json

```diff
@@ -932,6 +932,121 @@
+    },
+    {
+      "name": "auto_shorten_files_comment_links",
+      "input": "https://github.com/zulip/zulip/pull/16860/files#r539133612",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/pull/16860/files#r539133612\">zulip/zulip#16860/files#r539133612</a></p>"
+    },
+    {
+      "name": "auto_shorten_issues_links",
+      "input": "https://github.com/zulip/zulip/issues",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/issues\">https://github.com/zulip/zulip/issues</a></p>"
+    },
+    {
+      "name": "auto_shorten_pull_links",
+      "input": "https://github.com/zulip/zulip/pull",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/pull\">https://github.com/zulip/zulip/pull</a></p>"
+    },
+    {
+      "name": "auto_shorten_commit_links",
+      "input": "https://github.com/zulip/zulip/commit",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/commit\">https://github.com/zulip/zulip/commit</a></p>"
+    },
+    {
+      "name": "auto_shorten_clone_links",
+      "input": "https://github.com/zulip/zulip.git",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip.git\">https://github.com/zulip/zulip.git</a></p>"
+    },
+    {
+      "name": "auto_shorten_labels_links",
+      "input": "https://github.com/zulip/zulip/labels",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/labels\">https://github.com/zulip/zulip/labels</a></p>"
+    },
+    {
+      "name": "auto_shorten_labels_links",
+      "input": "https://github.com/zulip/zulip/labels/area%3A%20markdown",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/labels/area%3A%20markdown\">https://github.com/zulip/zulip/labels/area%3A%20markdown</a></p>"
+    },
+    {
+      "name": "auto_shorten_tree_links",
+      "input": "https://github.com/zulip/zulip/tree/chat.zulip.org",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/tree/chat.zulip.org\">https://github.com/zulip/zulip/tree/chat.zulip.org</a></p>"
+    },
+    {
+      "name": "auto_shorten_links",
+      "input": "https://github.com/marketplace/circleci",
+      "expected_output": "<p><a href=\"https://github.com/marketplace/circleci\">https://github.com/marketplace/circleci</a></p>"
+    },
+    {
+      "name": "auto_shorten_links",
+      "input": "https://github.com/foo/bar/baz",
+      "expected_output": "<p><a href=\"https://github.com/foo/bar/baz\">https://github.com/foo/bar/baz</a></p>"
+    },
+    {
+      "name": "auto_shorten_links",
+      "input": "https://github.com/foo/bar/baz/zar",
+      "expected_output": "<p><a href=\"https://github.com/foo/bar/baz/zar\">https://github.com/foo/bar/baz/zar</a></p>"
+    },
+    {
+      "name": "auto_shorten_links",
+      "input": "https://github.com/marketplace",
+      "expected_output": "<p><a href=\"https://github.com/marketplace\">https://github.com/marketplace</a></p>"
+    },
```

Comment:
Does having two with the same name work?  In any case, I think it'd be better to use unique names for readability, e.g. /`marketplace` here.

Reply:
> Does having two with the same name work? In any case, I think it'd be better to use unique names for
readability, e.g. `/marketplace` here.

@timabbott I completely didn't get this part. Can you please explain this a little more ?

Reply:
About half of these new tests have the `name` field set to `auto_shorten_links`, and the other half have unique names.  I think they should all have unique names.